### DISCUSSION TITLE
Compound variants

### DISCUSSION THREAD

**Discussion by la55u** [tamagui/tamagui#546] [Ideas]
Is something like this possible today or will be in the future? Compound variants would enable to style a component based on a combination of variants. For example:
```

const StyledButtonFrame = styled(ButtonFrame, {
   variants: {
    variant: {
      primary: { backgroundColor: "$purple" },
    },
    disabled: { true: { opacity: 0.9 } },
  },
  compoundVariants: {
    vars: [{ variant: "primary", disabled: true }],
    styles: {
      // all the styles from the above 2 vars + the styles defined here:
      backgroundColor: "$gray",
    },
  },
});
```

**Comment by natew** [comment]
IMO they are too complicated, I’d have to see some pretty convincing examples of how they save a lot of effort/code to afford the cost of having new terminology and runtime size.
